modules/src/mock_context.rs

```diff
@@ -74,6 +83,35 @@ impl MockContext {
         Self { clients, ..self }
     }
 
+    /// Similar to `with_client`, this function associates a client record to this context, but
+    /// additionally permits to parametrize two details of the client. If `client_type` is None,
+    /// then the client will have type Mock, otherwise the specified type. If
+    /// `consensus_state_height` is None, then the client will be initialized without any consensus
```

Comment:
This is an invalid context, when a client is created it must have a consensus state.

Reply:
Indeed, calling this results in an invalid context. We're using such an invalid context to construct conditions for a test to fail, [for instance here](https://github.com/informalsystems/ibc-rs/blob/d012f6c350ebd0f6ca3514a281039b5355fc12dc/modules/src/ics02_client/handler/create_client.rs#L146).

Reply:
I'm adding a remark to mention this!